projects/client/test/supplier/supplier.service.spec.ts

```diff
@@ -0,0 +1,170 @@
+      expect(result).toEqual(supplier);
+    });
+  });
+
+  describe('getAllSuppliers', () => {
+    it('should get all suppliers', async () => {
+      const suppliers: Supplier[] = [
+        {
+          id: '456',
+          email: 'test@test.com',
+          deliveryTime: 2,
+          countryId: '123',
+          country: {
+            id: '123',
+            name: 'Test Country',
+            tax: 10,
+            createdAt: String(new Date()),
+            updatedAt: String(new Date()),
+          },
+          createdAt: String(new Date()),
+          updatedAt: String(new Date()),
+        },
+      ];
+      const response = { data: suppliers };
+      (axios.get as jest.MockedFunction<typeof axios.get>).mockResolvedValue(response);
+
+      const result = await getAllSuppliers();
+
+      expect(axios.get).toHaveBeenCalledTimes(1);
+      expect(axios.get).toHaveBeenCalledWith('/api/suppliers');
+      expect(result).toEqual(suppliers);
+    });
+  });
+
+  describe('getSupplierById', () => {
+    it('should get a supplier by id', async () => {
+      const supplier: Supplier = {
+        id: '456',
+        email: 'test@test.com',
+        deliveryTime: 2,
+        countryId: '123',
+        country: {
+          id: '123',
+          name: 'Test Country',
+          tax: 10,
+          createdAt: String(new Date()),
+          updatedAt: String(new Date()),
+        },
+        createdAt: String(new Date()),
+        updatedAt: String(new Date()),
+      };
+      const response = { data: supplier };
+      (axios.get as jest.MockedFunction<typeof axios.get>).mockResolvedValue(response);
+
+      const result = await getSupplierById('456');
+
+      expect(axios.get).toHaveBeenCalledTimes(1);
+      expect(axios.get).toHaveBeenCalledWith('/api/suppliers/456');
+      expect(result).toEqual(supplier);
+    });
+  });
+
+  describe('updateSupplierById', () => {
+    it('should update a supplier by id', async () => {
+      const data: UpdateSupplierDto = {
+        email: 'test@test.com',
+        deliveryTime: 2,
+        countryId: '123',
+      };
+      const supplier: Supplier = {
+        id: '456',
+        email: 'test@test.com',
+        deliveryTime: 2,
+        countryId: '123',
+        country: {
+          id: '123',
+          name: 'Test Country',
+          tax: 10,
+          createdAt: String(new Date()),
+          updatedAt: String(new Date()),
+        },
+        createdAt: String(new Date()),
+        updatedAt: String(new Date()),
+      };
+      const response = { data: supplier };
+      (axios.patch as jest.MockedFunction<typeof axios.patch>).mockResolvedValue(response);
+
+      const result = await updateSupplierById('456', data);
+
+      expect(axios.patch).toHaveBeenCalledTimes(1);
+      expect(axios.patch).toHaveBeenCalledWith('/api/suppliers/456', data);
+      expect(result).toEqual(supplier);
+    });
+  });
+
+  describe('deleteSupplierById', () => {
```

Comment:
All describe unit tests are HAPPY path, but I would like to see negative tests....

Reply:
For example: 
What happens if I sent -1 as id, or 1000000 as id?

Reply:
Wait! It is not solved yet...